### DISCUSSION TITLE
Disk space check timed out in 5 seconds

### DISCUSSION THREAD

**Discussion by darabontors** [ericaltendorf/plotman#868] [Q&A]
I have an issue with Plotman archiving. I do have a large array of disks that probably take more than 5s to check free space on. Is there a way to change this timeout value to something much higher?

**Comment by darabontors** [comment]
In previous versions of Plotman I didn't get this message.

**Reply by altendky** [reply]
I have 18 disks on one of my systems and `df` executes in less than 10 milliseconds so I don't believe that's necessarily a limit.  Running remotely it takes about 1 second here.  Is this local or remote archiving?  What else are the plotter and farmer/harvester doing?  Have you tried manually running the disk space check command to see how long it takes?  You can find it in the general plotman log (configurable location).  It would look something like below.  If you add `time` on the front you can get a report of how long it takes to run.

```
ssh -p 22 altendky@server 'df -aBK | grep " /farm/sites/" | awk '\''{ gsub(/K$/,"",$4); printf "%s:%.0f\n", $6, $4*1024 }'\'''
```

Off the top of my head I don't know of recent changes that would affect this, but I might be forgetting something.  The 5 seconds is presently hardcoded so you would have to tweak the source.  It could be made configurable.

https://github.com/ericaltendorf/plotman/blob/3edef7ef97e0d4670322ae586823f4bbb1fc3c76/src/plotman/archive.py#L134

**Reply by darabontors** [reply]
I have 3 NetApp DS4246 disk shelves with 72 drives connected to 3 LSI external SAS controllers that are passed through physically to a VM with ample hardware resources. The VM is my farmer, I don't have any additional harvesters. It runs Ubuntu 20 LTS with Chia GUI as it farms and just the rsync daemon. My plotter is a 5900X, connected through 1GbE to the farmer. Both are on LAN. Plotter is running 3 instances of MadMax plotter with Plotman just archiving. 

Before the version 2 config came out, this timeout never appeared.
Local df -h takes ~ 5.8 seconds. 

How can I tweak the source file? (sorry, noob with github and compiling)
How do I install the modified python source file? How do I do this modification in plotman?
Thanks!

**Reply by darabontors** [reply]
Could I ask for this to be brought out into the config.yaml as a feature request?
